ignite/metrics/gan/inception_score.py

```diff
@@ -0,0 +1,115 @@
+
+    More details can be found in `Barratt et al. 2018`__.
+
+    __ https://arxiv.org/pdf/1801.01973.pdf
+
+
+    Args:
+        num_features: number of features predicted by the model or number of classes of the model. Default
+            value is 1000.
+        feature_extractor: a torch Module for predicting the probabilities from the input data.
+            It returns a tensor of shape (batch_size, num_features).
+            If neither ``num_features`` nor ``feature_extractor`` are defined, by default we use an ImageNet
+            pretrained Inception Model. If only ``num_features`` is defined but ``feature_extractor`` is not
+            defined, ``feature_extractor`` is assigned Identity Function.
+            Please note that the class object will be implicitly converted to device mentioned in the
+            ``device`` argument.
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            from ignite.metric.gan import InceptionScore
+            import torch
+
+            images = torch.rand(10, 3, 299, 299)
+
+            m = InceptionScore()
+            m.update(images)
+            print(m.compute())
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        num_features: Optional[int] = None,
+        feature_extractor: Optional[torch.nn.Module] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+
+        if num_features is None and feature_extractor is None:
+            num_features = 1000
+            feature_extractor = InceptionModel(return_features=False)
+
+        self._eps = 1e-16
+
+        super(InceptionScore, self).__init__(
+            num_features=num_features,
+            feature_extractor=feature_extractor,
+            output_transform=output_transform,
+            device=device,
+        )
+
+    @reinit__is_reduced
+    def reset(self) -> None:
+
+        self._num_examples = 0
+
+        self._prob_total = torch.zeros(self._num_features, dtype=torch.float64, device=self._device)
+        self._total_kl_d = torch.zeros(self._num_features, dtype=torch.float64, device=self._device)
+
+        super(InceptionScore, self).reset()
+
+    @reinit__is_reduced
+    def update(self, samples: torch.Tensor) -> None:
```

Comment:
Generated doc is wrong here. See parameters.
https://deploy-preview-2053--pytorch-ignite-preview.netlify.app/generated/ignite.metrics.inceptionscore#ignite.metrics.InceptionScore.update